File: back/src/main/java/firelib/common/core/Source.java
package firelib.common.core;

import firelib.domain.Ohlc;
import com.funstat.domain.InstrId;
import firelib.common.interval.Interval;
import kotlin.sequences.Sequence;

import java.time.LocalDateTime;
import java.util.List;

public interface Source{
    List<InstrId> symbols();
    Sequence<Ohlc> load(InstrId instrId);
    Sequence<Ohlc> load(InstrId instrId, LocalDateTime dateTime);
    String getName();
    Interval getDefaultInterval();
}

File: back/src/main/java/firelib/common/core/Source.kt
package firelib.common.core

import com.funstat.domain.InstrId
import firelib.common.interval.Interval
import firelib.domain.Ohlc
import java.time.LocalDateTime

interface Source {
    fun symbols(): List<InstrId>
    fun load(instrId: InstrId): Sequence<Ohlc>
    fun load(instrId: InstrId, dateTime: LocalDateTime): Sequence<Ohlc>
    fun getName(): String;
    fun getDefaultInterval(): Interval

    fun listen(instrId: InstrId, callback : (Ohlc)->Unit){}

}